MRG, MAINT: new website theme [skip github][skip azp][circle front]

drammock opened the pull request:
I've been poking away at the sphinx_pydata_theme off and on for the last few days.  Here's a WIP pull request that makes the switch.  All of you should feel free to jump in and push a commit or two if you see something I haven't fixed yet and know what the fix should be (if you don't know what the fix is, at least add it to the list below it so it doesn't get missed).

Once the circle build finishes you'll see there are some obvious remaining issues, but so far nothing that seems insurmountable. Here's a partial list:

- [x] ~~some CSS properties I've tried to override but aren't working (e.g., our "sidebar" callout boxes need a bgcolor or border to distinguish them from the main text)~~ *this turned out to be an issue with dev vs stable version of the theme: the CSS color variable names had changed*
- [x] ~~`navbar_align` property in conf.py isn't working~~ *only available in dev version of theme*
- [x] navbar has some weird spacing of items *opened https://github.com/pandas-dev/pydata-sphinx-theme/issues/302*
- [x] The homepage footer is offscreen / there is a bunch of whitespace after the carousel *this problem appears for @drammock locally but not on the circle build*
- [x] what to do with the homepage sidebar boxes (currently they aren't on the side and aren't boxed)
- [x] any page that still has a `.. contents::` directive should have it removed, since the theme includes page contents in the right sidebar
- [x] ~~haven't yet tried to incorporate sphinx_multiversion, so currently our version dropdown is missing.~~ *sphinx-multiversion is for **building** multiple versions all at once (which I don't think we want?) and in my local tests it failed to build anything older than 0.17.*

closes #8184
closes #8785

drammock commented:
here's the artifact link showing the new theme: https://24568-1301584-gh.circle-artifacts.com/0/dev/index.html

hoechenberger commented:
Nice work! Two remarks just after looking at the landing page:
<img width="1051" alt="Screen Shot 2021-01-17 at 11 09 56" src="https://user-images.githubusercontent.com/2046265/104837484-1fc10380-58b5-11eb-827f-0efc87d9f200.png">

👆this white-on-pink doesn't work for well for my eyes. Just checked with the Chrome Inspector, and indeed the contrast ration foreground / background is only about 1.3, whereas I believe accessibility guidelines demand a ratio of at least 4.5:1 for ordinary text:
<img width="545" alt="Screen Shot 2021-01-17 at 11 15 28" src="https://user-images.githubusercontent.com/2046265/104837509-5139cf00-58b5-11eb-93bf-df1ffb3d39d5.png">

The other thing I noticed was the large horizontal gap in the navigation bar. I do get what you're trying to achieve here, but somehow my mind found it slightly confusing, and I'd prefer to have the gap removed:
<img width="1051" alt="Screen Shot 2021-01-17 at 11 13 46" src="https://user-images.githubusercontent.com/2046265/104837544-978f2e00-58b5-11eb-8e4c-99f99439dfa9.png">


Lastly, are you planning to add optional Dark Mode support as well? That would be amazing now that also the forum has an automated color switcher :)

cbrnr commented:
👍 for removing the gap!

hoechenberger commented:
Hm, maybe that's already meant with this todo item:

> * [ ]  navbar has some weird spacing of items

drammock commented:
> this white-on-pink doesn't work for well for my eyes

This is one of the areas where CSS overrides aren't working in the way that I expect them to, is supposed to be the same color as on the current site. Same with the navbar above it, it's supposed to be dark gray



> maybe that's already meant with this todo item:

Well spotted

drammock commented:
> are you planning to add optional Dark Mode support as well?

Not in this PR. The switch to a well supported theme seems high priority in light of how our search functionality just broke unexpectedly, but dark mode is further down the list for me.

hoechenberger commented:
> This is one of the areas where CSS overrides aren't working in the way that I expect them to,

I haven't looked at the CSS, but sometimes a quick-and-dirty method once can try is adding `!important;` to a style definition, have you tried that?

agramfort commented:
I find the fonts a bit too small.

also when you look on a phone the first thing you see is the funding
support. It should be at the bottom.

>

cbrnr commented:
Things that others have not mentioned:

- The logo in the top left corner is tiny.
- We could put a link/logo to our MNE Discourse right next to GitHub and Twitter in the top right corner.
- It would be nice if the highlighted topics in the center image were clickable (and took you directly to the topic).
- The button "Installation - contents" should be called "Installation".
- The yellow background in classes (in the API section) doesn't really fit with the rest of the color scheme.
- The dash in the copyright statement (2012-2021) should be an ndash, i.e. 2012&ndash;2021.
- Maybe not for this PR, but I've always found it difficult to understand the difference between examples and tutorials. I assume it's probably the length (examples should be short whereas tutorials are supposed to be longer), but it makes it hard to find stuff when I have to decide between these categories as a first step.

hoechenberger commented:
> * Maybe not for this PR, but I've always found it difficult to understand the difference between examples and tutorials. I assume it's probably the length (examples should be short whereas tutorials are supposed to be longer), but it makes it hard to find stuff when I have to decide between these categories as a first step.

Good thing you mention this. I always thought it’s just me. I’ve always found it confusing, too. If **we’re** already struggling with this, how must (new) users feel?

drammock commented:
> sometimes a quick-and-dirty method is adding `!important;` to a style definition, have you tried that?

Yep, I was surprised that it didn't work.

> put a link/logo to our MNE Discourse right next to GitHub and Twitter

Great idea

> It would be nice if the highlighted topics in the center image were clickable

There is a small "try it ->" link on each one. I'll see about making the whole image clickable

> "Installation - contents" should be called "Installation"

Agree, there are a few page titles that look weird as next/prev links

> The yellow background in classes (in the API section) doesn't really fit with the rest of the color scheme

Not yet addressed because CSS cascading not working (see above). I think this might be a theme issue... Maybe style sheets added in conf.py are getting added before the theme default rules instead of after? Will look into it.

>  I've always found it difficult to understand the difference between examples and tutorials.

That is an issue I've been trying to chip away at for a while now, but progress is slow. My general intent is to make the tutorials long-form explanatory narratives that cover the most common use cases and are targeted at learners, and to make the examples into short-form how-to recipes that each demo only one thing (IMO these should be much fewer in number, and should focus on fairly specific advanced use cases like using blender to fix meshes, inverse imaging EEG with no MRI, etc).

drammock commented:
latest rendering of the new website theme: https://24648-1301584-gh.circle-artifacts.com/0/dev/index.html

agramfort commented:
https://pandas.pydata.org/docs/ is using larger fonts and full page width.
Is it an option?

>

hoechenberger commented:
I was also thinking that we should set the base size to 16px (currently 15px), and the ordinary text should be scaled to 1.25rem:

<img width="1440" alt="Screen Shot 2021-01-22 at 12 34 55" src="https://user-images.githubusercontent.com/2046265/105486103-3cdb4500-5cae-11eb-9358-c65b89ddd792.png">

Also the text should be justified and hyphenation enabled. You'll want to use sth like:

```
  word-break: break-word;  /* Chrome */
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  -o-hyphens: auto;
  hyphens: auto;
```
to make it work in ~all browsers.

hoechenberger commented:
> Also the text should be justified

Actually, the text on the front page could also be simply centered.

larsoner commented:
I would like to stay as close as possible to the default theme, so things like justification + hyphenation would fit better in the upstream repo. Also in this sense I agree that our devations from `pandas` in terms of font (I think?) and font size (at least in the navbar) should be removed. (It's possible they have custom CSS for this but I'm assuming they don't.)

jorisvandenbossche commented:
Sorry to enter your discussion like, but I was checking this out to investigate the navbar bug report that @drammock opened (pandas-dev/pydata-sphinx-theme#302, it's a bug in the theme, but there is an easy workaround, see the issue report). So thought I could directly give some comments/replies here as well :)

> are you planning to add optional Dark Mode support as well?

There is WIP PR to add dark mode to the theme: https://github.com/pandas-dev/pydata-sphinx-theme/pull/273 (but it's certainly not yet ready, contributions are welcome!)

> Maybe style sheets added in conf.py are getting added before the theme default rules instead of after? Will look into it.

Looking in the browser dev tools, the custom css gets loaded after the theme css, so that shouldn't be the problem normally (also in geopandas we have some custom css which is working there)

> The logo in the top left corner is tiny

In the master version of the theme, we removed some padding around the logo, so it should appear larger now. If you still want to have it larger, there is now a ``--header-height`` variable that can be set to get this.

> The yellow background in classes (in the API section) doesn't really fit with the rest of the color scheme

I think this is something we inherit from sphinx itself, and I agree it can look a bit off (also for other sites where we are using this theme, so it might be a welcome change upstream in the pydata-sphinx-theme as well)

> I find the fonts a bit too small.

In the development version of the theme, we are adding CSS variables so that it should be easy to make the font size larger

> our devations from pandas in terms of font (I think?) and font size (at least in the navbar) should be removed

Note that with the CSS variables (I see that for the font @drammock already used it here), it's very easy to customize this a bit, and those variables are added exactly so downstream users *can* do this (without needing the know the exact CSS classes etc that need to be overridden). Now, of course, you are for sure free to also follow the base theme ;), I only want to mention that those variables are meant to be overridden in a robust way.

Related to this, we are also planning to drop the custom fonts we now include in the base theme (eg OpenSans and Lato), and use the system font stack instead as default (https://github.com/pandas-dev/pydata-sphinx-theme/pull/285). But with the CSS variable it should be easy to still use another font if desired.

---

On another note, if you would like to keep the navbar with a dark color, like the current mne website, that should also be possible. Eg MetPy recently did this, see eg https://unidata.github.io/MetPy/latest/userguide/startingguide.html

larsoner commented:
Thanks for the feedback @jorisvandenbossche ! To me the goal is to deviate as little as possible from the base theme. Do you know offhand how much custom CSS pandas for example uses? Based on looking at [doc/source/_static/css](https://github.com/pandas-dev/pandas/tree/master/doc/source/_static/css) it looks like the answer is "for most pages, little to none"

jorisvandenbossche commented:
Yes, in pandas we don't override much, but that's probably also because we initially started the theme for pandas. Eg for geopandas we override some color elements: https://github.com/geopandas/geopandas/pull/1666/files (and the PR shows how using CSS variables makes this much cleaner) 
For example, currently the "highlight" color for active nagivation elements is the purple of the pandas logo. I actually have been planning to make this a more "neutral" color in the base theme, and then override this to be this purple only in the pandas docs.

larsoner commented:
Looks like great progress, and +1k/-3.9k is awesome! Minor list of things that I think would get it MRG+1 from me:

- [ ] Footer institutions not aligned center
- [ ] "Download Python source code" CSS yellow does not match theme well, maybe use some named CSS theme color?- [ ] Version drop-box color is red, should probably match the pink of the "These are dev docs" warning box?
- [ ] Our navbar spacing does not match pandas
- [ ] Docs search is in the wrong place relative to `pandas` (theirs is upper left in meta-TOC or whatever, ours is upper right in navbar)
- [ ] Our fonts do not match pandas
- [ ] Tiny logo should be improved in size and readability, ![](https://24674-1301584-gh.circle-artifacts.com/0/dev/_static/mne_logo_small_gray.svg) vs <img src="https://pandas.pydata.org/docs/_static/pandas.svg" width="88">

Restructuring so that we can get the left-content-right pattern more consistent would be good, like:

https://pandas.pydata.org/docs/getting_started/index.html
https://pandas.pydata.org/docs/user_guide/dsintro.html

vs places we only have a left bar (no `contents` / sections in a given document):

https://24674-1301584-gh.circle-artifacts.com/0/dev/install/index.html

and places we have no left bar (not part of a toctree or whatever presumably):

https://24674-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/index.html
https://24674-1301584-gh.circle-artifacts.com/0/dev/overview/implementation.html#implementation

or neither even though we could probably have both

https://24674-1301584-gh.circle-artifacts.com/0/dev/overview/index.html

I think this would potentially be a big usability / nav benefit! I think this could be done in subsequent PRs if you want, though.

larsoner commented:
+1.2k -4.1k 😍 

I'll merge with `master` so that it builds and we can see it but feel free to push-force right over it with a rebase if you want

drammock commented:
[Latest rendering here](https://24761-1301584-gh.circle-artifacts.com/0/dev/index.html). This is ready for a thorough look from the devs, and suggestions on how to deal with the remaining issues:

1. Fonts are still small.  Overriding the `--font-size-base` CSS variable doesn't seem to have any effect (and in general, the variables that the theme is supposed to define like `--color-link` don't seem to be available).
2. On Firefox, on any page that does not need vertical scrolling, there is a lot of whitespace between the next/prev button and the footer and the footer ends up just offscreen. Doesn't happen on Chromium.
3. The right sidebar does not appear at all on narrow screens / mobile.  This is mostly OK since it only contains "page contents" links, but for the homepage it means that the "version box" doesn't appear (which is currently the only place where the "cite" link exists).
4. The homepage's "direct financial support" box feels a bit more prominent than I would like. Maybe two side-by-side cards (one for the version box and one for the funders) would fix both this and the previous problem?
5. The [cited page](https://24761-1301584-gh.circle-artifacts.com/0/dev/cited.html) is not linked from anywhere (?) and hence not discoverable.
6. On desktop, the collapsible divs for different OSes don't auto-open to the detected OS.  On mobile (android), the Linux one starts open and can't be closed (trying to close it, it will re-open itself immediately).
7. It would be nice if the [glossary](https://24761-1301584-gh.circle-artifacts.com/0/dev/glossary.html) had right-sidebar within-page links. It requires getting the glossary entries to get treated like toctree elements, or else faking that in some way (inserting hidden section headings before each glossary entry?)  I consider this "nice to have, not a blocker".
8. This theme has the same problem our old theme had: within-page anchors are often hidden under the top navbar (so if I click on a backreference link from the References section on [this page](https://24761-1301584-gh.circle-artifacts.com/0/dev/overview/implementation.html), I then have to scroll up the page a bit to actually see where the in-text citation is).  I will note however that this does not seem to be a problem with the within-page headings, so that's an improvement over the old theme.  Probably needs an upstream fix.
9. The API reference page has been split by section, so that it is navigable using the left sidebar. I think this is a usability win, but I don't love how the API landing page looks now.  I think it would be possible to make the toctree hidden, and stick in a bunch of `.. include::` directives, so that the landing page still had everything, but the navbar would take you to the individual pages (downside: content is duplicated, need to be careful about link targets not getting duplicated too).
10. Another thing about the API landing page (was true in old site too): the inclusion of `:py:mod:mne` and the `autodoc` directive just below the intro text results in a self-linking `mne:` and then the module description "MNE software for MEG and EEG data analysis" appearing on the page. Can we suppress these somehow?  Or alternatively, actually leverage autodoc to generate the API reference instead of manually sorting things by category?  (bigger question possibly for a separate PR).

Already fixed locally, not yet pushed:

1. Section order on the datasets overview page (someone put the "refmeg noise" section *after* the footbibliography)

larsoner commented:
> The right sidebar does not appear at all on narrow screens / mobile. This is mostly OK since it only contains "page contents" links, but for the homepage it means that the "version box" doesn't appear (which is currently the only place where the "cite" link exists).

Indeed I see this as a blocker. 

But this also gave me an idea -- put funding sources in the left column (or right maybe?), and give it this property that it disappears below some size (maybe sm-something). I don't think we need to show those on mobile. Basically if you solve the disappearing-right-column problem you might have the solution to the funding-sources-too-prominent problem.

> The cited page is not linked from anywhere (?) and hence not discoverable.

I would put "How to cite" in bottom of the left bar of Documentation, and then add this as "Papers citing MNE" beneath that. Then both are in some toctree, and how-to-cite is both on the landing page and documentation page.

> On desktop, the collapsible divs for different OSes don't auto-open to the detected OS.

This is a regression so should be fixable with appropriate CSS/JS somehow. Let me know if you want me to look, I think I had a hand in implementing the collapasing ones on `main`.

> This theme has the same problem our old theme had: within-page anchors are often hidden under the top navbar

Did you try some variant of https://stackoverflow.com/questions/10336194/top-nav-bar-blocking-top-content-of-the-page#answer-10336221? I think I tried this with our current theme at some point but maybe using Bootstrap 4 will make it more fixable.

> Can we suppress these somehow?

Can you `.. container:: d-none` them?

For everything else I see them as minor issues we can fix in subsequent PRs.

drammock commented:
> > The right sidebar does not appear at all on narrow screens / mobile. This is mostly OK since it only contains "page contents" links, but for the homepage it means that the "version box" doesn't appear (which is currently the only place where the "cite" link exists).
> 
> Indeed I see this as a blocker.
> 
> But this also gave me an idea -- put funding sources in the left column (or right maybe?), and give it this property that it disappears below some size (maybe sm-something). I don't think we need to show those on mobile. Basically if you solve the disappearing-right-column problem you might have the solution to the funding-sources-too-prominent problem.

DONE

> > The cited page is not linked from anywhere (?) and hence not discoverable.
> 
> I would put "How to cite" in bottom of the left bar of Documentation, and then add this as "Papers citing MNE" beneath that. Then both are in some toctree, and how-to-cite is both on the landing page and documentation page.

DONE

> > On desktop, the collapsible divs for different OSes don't auto-open to the detected OS.
> 
> This is a regression so should be fixable with appropriate CSS/JS somehow. Let me know if you want me to look, I think I had a hand in implementing the collapasing ones on `main`.

FIXED (I think; need to test on mobile once circle build finishes)

> > This theme has the same problem our old theme had: within-page anchors are often hidden under the top navbar
> 
> Did you try some variant of https://stackoverflow.com/questions/10336194/top-nav-bar-blocking-top-content-of-the-page#answer-10336221? I think I tried this with our current theme at some point but maybe using Bootstrap 4 will make it more fixable.

FIXED-ish.  Ended up adapting the JS from [here](https://github.com/copyleft-org/copyleft-guide/commit/476a42bf0d737e13a561dbaf6f4e1e91a333e80d). I probably put it in a dumb place (header of `_templates/layout.html`) so LMK where it should really go.

> > Can we suppress these somehow?
> 
> Can you `.. container:: d-none` them?

DONE (didn't know about container directives, cool)

> For everything else I see them as minor issues we can fix in subsequent PRs.

YAY!

drammock commented:
Latest rendering: https://24797-1301584-gh.circle-artifacts.com/0/dev/index.html

cbrnr commented:
This is starting to look really nice! I've noticed three issues that might not have been mentioned before:

1. The institutions bottom bar is always invisible, which means I need to scroll down even if I make my browser window as high as possible. I think it would be nice if it was visible (in case there is enough vertical space).
2. The navigation buttons with their outline and the `>>` symbols look a bit awkward. I guess replacing `>>` with a nice unicode arrow would improve their look.
3. Highlighting the left/right corners in the carousel doesn't quite work, the highlight doesn't disappear if I move the mouse cursor away - see screen capture below.

https://user-images.githubusercontent.com/4377312/106101142-c15e1580-613d-11eb-8e4f-10ff753158a9.mov

drammock commented:
> * The institutions bottom bar is always invisible, which means I need to scroll down even if I make my browser window as high as possible. I think it would be nice if it was visible (in case there is enough vertical space).

Yep, this was # 2 in my [list of remaining issues](https://github.com/mne-tools/mne-python/pull/8757#issuecomment-768429550)

> * The navigation buttons with their outline and the `>>` symbols look a bit awkward. I guess replacing `>>` with a nice unicode arrow would improve their look.

Good idea, will try.

> * Highlighting the left/right corners in the carousel doesn't quite work, the highlight doesn't disappear if I move the mouse cursor away - see screen capture below.

I can reproduce. Will see if I can fix it.

cbrnr commented:
@drammock your issue 2 happened on Chrome (macOS).

larsoner commented:
I like the new "carousel"! I would say make all tiles the same size, and maybe change `time_dilation` in the dSPM example. It seems a bit frantic now. I can push commits for this if you want.

drammock commented:
I'm off duty for the next few days so feel free to push commits here. (Set aside a few days to do "real science")

drammock commented:
Personally I was thinking to change alpha to 0.2, make text fully disappear on hover, ditch the blue buttons to make the whole thing clickable, fix the height at 225 or so, and add a gutter separating the tiles (margin won't work, it messes up the flex wrapping). But feel free to just take that as a starting point and experiment.

larsoner commented:
Well when you put it that way... I'm happy to wait a few days for you to look into it :)

drammock commented:
> Well when you put it that way... I'm happy to wait a few days for you to look into it :)

😆 Didn't mean to be a control freak; those are just the ideas I played around with in the browser console last night that I didn't have a chance to fully test / implement. No prob if you'd rather wait for me to do it though

larsoner commented:
No I like your plan, and at the same time I'm glad that I won't have to fight with the CSS and JS to make it happen :boxing_glove:

jorisvandenbossche commented:
> 1. Fonts are still small.  Overriding the `--font-size-base` CSS variable doesn't seem to have any effect (and in general, the variables that the theme is supposed to define like `--color-link` don't seem to be available).

That's because the variables are not yet available in the last release, but only in the master version. 
(I would personally suggest to directly use the latest master version, and you can still pin to the latest commit to have a pinned version of the theme, but then you can directly make use of this).

> 7\. It would be nice if the [glossary](https://24761-1301584-gh.circle-artifacts.com/0/dev/glossary.html) had right-sidebar within-page links. It requires getting the glossary entries to get treated like toctree elements, or else faking that in some way (inserting hidden section headings before each glossary entry?)  I consider this "nice to have, not a blocker".

That's probably not that easy, but as a starter, those entries would need to be "targets" (like places where you put a label so you can create an internal reference with anchors), which is at the moment also not yet the case.

> > 8\. This theme has the same problem our old theme had: within-page anchors are often hidden under the top navbar
>
> FIXED-ish. Ended up adapting the JS from [here](https://github.com/copyleft-org/copyleft-guide/commit/476a42bf0d737e13a561dbaf6f4e1e91a333e80d). I probably put it in a dumb place (header of `_templates/layout.html`) so LMK where it should really go.

This is actually fixed in the latest master version of the theme, using a css-only fix (it was especially to non-header links (eg place in the API docs) that it was not yet working). (https://github.com/pydata/pydata-sphinx-theme/pull/283)

> 2\. The navigation buttons with their outline and the `>>` symbols look a bit awkward. I guess replacing `>>` with a nice unicode arrow would improve their look.

Feel free to upstream a fix, I agree it's not the most fancy design at the moment (also the box around it can be improved I think)

drammock commented:
> That's because the variables are not yet available in the last release, but only in the master version.

Thanks. We can pin to a specific commit in master, access to those theme variables will make some things much easier for us.

> [glossary in right sidebar] That's probably not that easy

Thanks for confirming my suspicion here.

> [anchors hidden below top nav] is actually fixed in the latest master version of the theme

Awesome. Another reason to pin to master.

> [nex/prev links] Feel free to upstream a fix

Will do!

drammock commented:
We'll see how the circle build looks, but apart from waiting for a fix on https://github.com/pydata/pydata-sphinx-theme/issues/304 I think this might be ready to go.  I put in a workaround for https://github.com/pydata/pydata-sphinx-theme/issues/315 already.

drammock commented:
> I think this might be ready to go

I spoke too soon.  The javascript for fixing the anchor scrolling [isn't working](https://24958-1301584-gh.circle-artifacts.com/0/dev/overview/implementation.html#equation-matrix-form) and the homepage gallery tiles need some TLC in order to look good at all browser widths [rendered homepage](https://24958-1301584-gh.circle-artifacts.com/0/dev/index.html)

drammock commented:
Here's the latest rendering: https://24990-1301584-gh.circle-artifacts.com/0/dev/index.html

This is still waiting for resolution of two theme issues:
- https://github.com/pydata/pydata-sphinx-theme/issues/304 (that problem also affects our version dropdown, but their WIP solution fixes it)
- https://github.com/pydata/pydata-sphinx-theme/issues/315 (my attempted javascript workaround seems to be effective if I run it in the browser console, but not if it's loaded by the page... I'm holding out hope that they find an upstream CSS-based fix)

Besides that, I think the only remaining TODO relates to the frontpage gallery: the images that we show in each tile are very different sizes/aspect ratios, which is causing some minor problems:

- the zoom effect is a little unpredictable at different screen sizes
- the 3 images with black backgrounds can look bad if they get narrower than the container, and because the container size can change with screen width, it's hard to pick a zoom level that both looks good and is guaranteed not to be smaller than the container.
 
A possible solution is to be a little more intentional about crafting the images that we want to use on each tile (i.e., either tweaking the tutorials they come from to make them more uniform aspect; avoiding black backgrounds, etc), or possibly making one new "example" where all it does is generate the 6 images we want for our homepage gallery.  That way they can be as bespoke as we want without it distracting from the example/tutorial content, and in theory it might speed up `circle front` because it would only need to run a single example with no extraneous plots.  Other suggestions are welcome.

hoechenberger commented:
Hi Dan, thanks for your work on this!

The rendering doesn't look good on my 10.5" iPad in portrait mode, and tapping the burger button doesn't show the menu. Things look nice in portrait mode, however.

Here's a screen recording:

https://user-images.githubusercontent.com/2046265/106860864-45c41180-66c5-11eb-938d-4d9bf7e7d18d.mp4

hoechenberger commented:
This behavior can be reproduced in the Chrome Inspector if you select the "iPad" device
<img width="853" alt="Screen Shot 2021-02-04 at 08 47 51" src="https://user-images.githubusercontent.com/2046265/106861152-ad7a5c80-66c5-11eb-80e7-f600c56d1e0b.png">

drammock commented:
The broken hamburger menu is known theme bug 304, linked to above. The off-center look in portrait mode could probably be fixed by changing the threshold at which we hide the homepage sidebars... I'll look into it.

drammock commented:
@hoechenberger I think I've fixed your issue with the iPad: https://25004-1301584-gh.circle-artifacts.com/0/dev/index.html
I had to make the specific grant names/numbers disappear on narrower screens, but to me it seems fine, WDYT?

The "statistics" gallery tiles is mysteriously mostly black now (can't have been any change in this PR I don't think?).  After that I'll see if I can speed up getting fixes for the theme issues we're waiting on.

larsoner commented:
The version dropdown is broken for me, this is it "open":

![Screenshot from 2021-02-05 09-52-02](https://user-images.githubusercontent.com/2365790/107049236-d8cd7c00-6797-11eb-9d68-c79079ae4c88.png)

The stats example is not your problem, it's really generated that way:

https://25004-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/stats-source-space/plot_stats_cluster_spatio_temporal.html

@GuillaumeFavelier can you replicate locally? Is it from the recent PR where we messed with `paintEvent`?

drammock commented:
> The version dropdown is broken for me

@larsoner this was mentioned already:

> This is still waiting for resolution of two theme issues:
> 
>     * [pydata/pydata-sphinx-theme#304](https://github.com/pydata/pydata-sphinx-theme/issues/304) (that problem also affects our version dropdown, but their WIP solution fixes it)

The upstream fix is not yet merged.

cbrnr commented:
I still think that the sponsor sidebar on the left is way too prominent. Pandas doesn't do this - why did we decide to put it there? I'd rather put the focus on the content (the carousel) and move the sponsors below it. You mentioned that this would be one option for the mobile site, but I'd do it for all versions/sizes.

drammock commented:
> I still think that the sponsor sidebar on the left is way too prominent. 

"still"?  I missed where you mentioned this before. Does anyone else agree?

To me, it looks strange if the only thing in the homepage left sidebar is the "quick links" version box.  I also feel that funders is not "main" content, so it belongs in a sidebar rather than in the main window.  I'll see what I can do to make it stand out less...  maybe making the text and/or logos a little smaller.

larsoner commented:
On the main pandas landing page their primary fiscal sponsors are on first page of content at least for desktops:

https://pandas.pydata.org/

![Screenshot from 2021-02-05 10-53-17](https://user-images.githubusercontent.com/2365790/107056591-7cbb2580-67a0-11eb-9522-0a77445c8b2d.png)

For us if we move it below the gallery images it's off the first page. +1 for just shrinking the text/logos as necessary instead of moving them. I played around and reducing the font-weight and font-size seemed to be okay. For example adding `font-weight: 200` at the UL level makes things less prominent (main text ends up 200, and names end up bigger, probably 400 somewhere because of the <strong>).

I don't see the NIH grants listed, though, which we should fix :)

cbrnr commented:
> "still"? I missed where you mentioned this before. Does anyone else agree?

True, I never mentioned it before, but I always felt that way 😄. Sorry to bring this up so late in the process, and like I said you can ignore it exactly because I didn't mention it before.

However, the pandas website has a much nicer look because their sponsors are not the first thing you see. Maybe making the text smaller *and* moving the sidebar to the right would be an option for us too? Then at least the first thing people read is MNE and not version links and sponsors.

drammock commented:
> I don't see the NIH grants listed, though, which we should fix :)

I inadvertantly deleted that line when editing the file.  Will be restored when I push the next batch of commits.

drammock commented:
latest rendering: https://25029-1301584-gh.circle-artifacts.com/0/dev/index.html

I've reduced the funders sidebar prominence, restored the accidentally deleted NIH line, fixed the offscreen footer problem for all screen widths, and got the javascript approach to anchors-covered-by-topnav working.

@jorisvandenbossche you can see the javascript solution to the inline anchor target issue pretty clearly on [this page](https://25029-1301584-gh.circle-artifacts.com/0/dev/overview/implementation.html#equation-legendre) since it has several numbered equations and citation backreferences.  The link above should take you right to equation 2 (and it won't be hidden under the navbar).  I can put it in a PR to the theme if you like how it works.

At this point I think all that's missing is a theme fix for the topnav `overflow: hidden` problem.  I can hack in an override in our CSS but I think https://github.com/pydata/pydata-sphinx-theme/pull/313 is close to being merged (?) so I'd say let's just wait for that.

larsoner commented:
Beautiful @drammock ! The sidebar prominence is much better. For some reason (HiDPI macOS maybe?) The funder is smaller than the grant number(s), which seems backward:

<img width="299" alt="Screen Shot 2021-02-05 at 6 26 39 PM" src="https://user-images.githubusercontent.com/2365790/107099477-ca0ab780-67df-11eb-8615-e825f9bdd31a.png">

larsoner commented:
> EDIT oh, and we should fix the bizarre "statistics" gallery figure before merging this one in :)

It's there and even worse on https://mne.tools/dev/:

https://mne.tools/dev/_images/sphx_glr_plot_stats_cluster_spatio_temporal_001.png

So this is my and @GuillaumeFavelier's problem, 95% sure we introduced this by hacking `paintEvent` recently. I wouldn't wait for it in this PR.

GuillaumeFavelier commented:
😱

drammock commented:
> For some reason (HiDPI macOS maybe?) The funder is smaller than the grant number(s), which seems backward

I can reproduce. I think it's asking for weight 300, which is the "light" weight, and it's not finding it, so it's falling back to a variable-weight version and scaling it.  I'll try the fix locally, and if it works, commit the light weight files to the repo

cbrnr commented:
@drammock just to make sure, you didn't mention the hamburger menu not working in your all that's missing summary - I guess we're waiting for an upstream fix, right?

The sponsors sidebar is better, but can we vote on moving it to the right instead?

drammock commented:
> you didn't mention the hamburger menu not working

That is what this refers to: "At this point I think all that's missing is a theme fix for the topnav `overflow: hidden` problem."

> The sponsors sidebar is better, but can we vote on moving it to the right instead?

I'll cook up two screenshots later today so people know what they're voting on.

drammock commented:
OK, here are 2 screenshots for folks to weigh in on.  Current state (funders box on left side of homepage):

![left-side](https://user-images.githubusercontent.com/1810515/107570841-afee2200-6baf-11eb-8aed-430272e73b7a.png)

@cbrnr's request (funders box on right side).

![right-side](https://user-images.githubusercontent.com/1810515/107570860-b67c9980-6baf-11eb-8150-9c3b23fa0727.png)

cbrnr commented:
TBH I'd also move the other box to the right.

drammock commented:
Here is @cbrnr's other request: both boxes on the right
![both-right](https://user-images.githubusercontent.com/1810515/107574131-f5ace980-6bb3-11eb-808f-a43ee54f5755.png)

larsoner commented:
Looking at all three I prefer all-left

cbrnr commented:
All right doesn't look great because the left sidebar still takes up space. I'd remove it completely so that the main content moves to the left. Basically how the pandas website does it.

drammock commented:
OK, with no left sidebar: 
![no-left-sidebar](https://user-images.githubusercontent.com/1810515/107574854-c21e8f00-6bb4-11eb-8c0f-e371652699e4.png)

Remember that we are working within a theme and trying to strike a balance between using a stock theme (less to maintain, less chance of conflict with upstream theme improvements) and tweaking the theme to suit our brand/content.  To me the best compromise as far as the homepage goes is to use the left sidebar and ignore the right one (because left one is wider) rather than the other way around.

cbrnr commented:
I actually like that best (the one without the left sidebar). It's also what pandas uses, right?

<img width="2394" alt="Screen Shot 2021-02-11 at 09 32 06" src="https://user-images.githubusercontent.com/4377312/107615320-3a845f00-6c4c-11eb-9141-3512c66aabdf.png">

drammock commented:
> I actually like that best (the one without the left sidebar). It's also what pandas uses, right?

No.  Pandas' homepage diverges significantly from the pydata theme (they have [a separate, bespoke template](https://github.com/pandas-dev/pandas/blob/master/doc/source/index.rst.template) for the homepage)

cbrnr commented:
I didn't know that. I agree that sticking as close as possible to the template is important, and if this means having two boxes in the left sidebar that's OK.

Speaking of the top box, I don't like that it contains two links that are also present in the top menu. On the other hand, a search box is missing.

agramfort commented:
To use as much space as possible without deviation from thème I would put toc on the left and funding on the right.

larsoner commented:
I can live with @agramfort's split -- I say we do that, wait for upstream PRs to be merged, then merge this (and make further tweaks in additional PRs) !

drammock commented:
- Azure pip-pre failure is #8873 and @GuillaumeFavelier already has a fix in place (not yet merged).
- LGTM failure is a mystery to me (`[build-err] Only found JavaScript or TypeScript files that were empty or contained syntax errors` but it doesn't say which files are unsyntactic)... AFAICT the only javascript that I *added* in this PR is working correctly.
- latest rendering: https://25174-1301584-gh.circle-artifacts.com/0/dev/index.html

cbrnr commented:
Looking very good @drammock! I like it! Only some minor points should be addressed, but this could also happen in another PR:

- A dedicated sponsors page would be nice (see how pandas does it) - we could include a link to it in the left sidebar, and then we can discuss what to do with the right sidebar (remove, keep but link the title to the sponsors page, show the link in the left sidebar only when the right sidebar disappears).
- Some numbers/characters in the sponsors sidebar are still different from the main text (e.g. P41, MIND, IDEX).
- There are still duplicate links in the left sidebar and the top nav (Docs, Get help); I'd only keep them in the top navbar.
- My preference was having all sidebars on the right; people didn't like this, and now we have two sidebars, which I think looks OK, but the problem is that the right sidebar disappears below a certain width with no replacement. This is not ideal. If we remove two links from the Version box (see previous bullet point), the sponsors sidebar could go below the version sidebar on the left.

cbrnr commented:
One more thing that keeps annoying me: Often, searching for a particular answer on Google takes me to the dev version of our website. If I switch to the stable version, it always takes me to the homepage instead of the corresponding page in stable. Is this something that could be solved?

larsoner commented on mne/tests/test_docstring_parameters.py:
```diff
@@ -275,19 +276,23 @@ def test_documented():
 
-    doc_file = op.abspath(op.join(op.dirname(__file__), '..', '..', 'doc',
-                                  'python_reference.rst'))
+    doc_dir = op.abspath(op.join(op.dirname(__file__), '..', '..', 'doc'))
+    api_dir = op.join(doc_dir, 'api')
+    doc_file = op.join(doc_dir, 'python_reference.rst')
```
These changes can probably be reverted now?

drammock replied:
not simply reverted, as we now have to check 22 separate files instead of just `python_reference.rst`

larsoner commented on mne/tests/test_docstring_parameters.py:
```diff
@@ -1,5 +1,6 @@
 import inspect
 from inspect import getsource
+import os
```
```suggestion
```

larsoner commented on doc/conf.py:
```diff
@@ -97,73 +97,9 @@
-# source_encoding = 'utf-8-sig'
-
-# The main toctree document.
+# The master toctree document.
```
```suggestion
# The main toctree document.
```

larsoner commented:
Thanks @drammock !!!

hoechenberger commented:
🎈
